x/feegrant/types/msgs_test.go

```diff
@@ -0,0 +1,69 @@
+package types_test
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/feegrant/types"
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+)
+
+func TestMsgs(t *testing.T) {
+	addr, _ := sdk.AccAddressFromBech32("cosmos1aeuqja06474dfrj7uqsvukm6rael982kk89mqr")
+	addr2, _ := sdk.AccAddressFromBech32("cosmos1nph3cfzk6trsmfxkeu943nvach5qw4vwstnvkl")
+	atom := sdk.NewCoins(sdk.NewInt64Coin("atom", 555))
+	basic := &types.BasicFeeAllowance{
+		SpendLimit: atom,
+		Expiration: types.ExpiresAtTime(time.Now().Add(3 * time.Hour)),
+	}
+	cases := map[string]struct {
+		grantee sdk.AccAddress
+		granter sdk.AccAddress
+		grant 	*types.BasicFeeAllowance
+		valid 	bool
+	}{
+		"valid":{
```

Comment:
event though that's already tested elsewhere, could you add test cases for invalid granter and grantee for both messages?
Also I believe having two separate tests for both msgs types would be clearer.

Reply:
> Also I believe having two separate tests for both msgs types would be clearer.

[moved out](https://github.com/cosmos/cosmos-sdk/pull/9193/files#diff-10a1e3f076e16194dab57477c71f3ca8a99549c1f55815392b5d26957e18bbf1R77-R128)

>event though that's already tested elsewhere, could you add test cases for invalid granter and grantee for both messages?

[okay added some invalid granter/grantee cases](https://github.com/cosmos/cosmos-sdk/pull/9193/files#diff-10a1e3f076e16194dab57477c71f3ca8a99549c1f55815392b5d26957e18bbf1R34-R44)